porch/pkg/engine/engine.go

```diff
@@ -455,13 +469,30 @@ func (m *updatePackageMutation) Apply(ctx context.Context, resources repository.
 		return repository.PackageResources{}, nil, err
 	}
 
-	ref := m.task.Clone.Upstream.Git.Ref
-
-	// TODO: This is a hack
-	packageName := filepath.Base(m.task.Clone.Upstream.Git.Directory)
-	packageName = strings.TrimPrefix(packageName, ".git")
+	var ref string
+	var packageDir string
+	upstream := m.task.Clone.Upstream
+	switch upstream.Type {
+	case api.RepositoryTypeGit:
+		ref = upstream.Git.Ref
+		packageName := filepath.Base(m.task.Clone.Upstream.Git.Directory)
```

Comment:
Naive question: is it always true that the packageName is the same as the directory? Is that still true with subpackages?

Reply:
Just a note that this code goes away with my PR. My PR ensures that update is supported only for porch-registered packages.